### PR TITLE
Add 2021 Community Survey Results Blog

### PR DESCRIPTION
**Pull request by rylev**
This adds a high-level analysis of the results of the [2021 Community Survey](https://blog.rust-lang.org/2021/12/08/survey-launch.html).

[rendered](https://github.com/rylev/blog.rust-lang.org/blob/rust-survey-2021-results/posts/2022-02-15-Rust-Survey-2021.md)

### CODE REVIEW

File: posts/2022-02-08-Rust-Survey-2021.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+The survey was offered in 10 languages with 78% filling out the survey in English followed by Chinese (6%), German (4%), and French (3%). Despite English being the language most respondents completed the survey in, respondents hailed from all around the world. The United States was the country with the largest representation at 24% followed by Germany (12%), China (7%), and the U.K. (6%). In total 113 different countries and territories were represented through this survey!
+
+English, however, is not the language of choice for all Rustaceans with nearly 7% preferring not to use English as a language for technical communication. An additional 23% of respondents prefer another language in addition to English. The most common prefered languages (besides English) follow where Rustaceans live: Chinese, German, and French being the top 3. Japanese, Chinese, and Russian speakers were the most likely to prefer not to use English for technical conversation.
+
+## Using Rust
+
+The percentage of people using Rust continues to rise, with 81% of those currently using Rust using it on at least a weekly basis compared to 72% from last year's survey.
```
Comment (flip1995) [inline posts/2022-02-08-Rust-Survey-2021.md:23]: This sentence is really hard to parse for me (English is not my first language). Maybe:

```suggestion
The percentage of people using Rust continues to rise. Of those using Rust, 81% are currently using it on at least a weekly basis compared to 72% from last year's survey.
```

File: posts/2022-02-08-Rust-Survey-2021.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+75% of all Rust users say they can write production ready code, though 27% say that it is at times a struggle.
+
+![Rust expertise](/images/2022-02-RustSurvey/expertise.png)
+
+Overall, Rustaceans seem to be having a great time writing Rust with only 1% saying isn't fun to use. Only a quarter of a percent find Rust doesn't have any real benefit over other programming languages.
```
Comment (flip1995) [inline posts/2022-02-08-Rust-Survey-2021.md:29]:
```suggestion
Overall, Rustaceans seem to be having a great time writing Rust with only 1% saying it isn't fun to use. Only a quarter of a percent find Rust doesn't have any real benefit over other programming languages.
```

File: posts/2022-02-08-Rust-Survey-2021.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Rust at Work
+
+Rust can now safely be classified as a language used by people in professional settings. 59% of respondents who currently use Rust, use it at least occasionally at work with 23% using Rust for the majority of their coding. This is a large increase over last year where 58% of respondents didn't use Rust at work compared with this year's 41%.
+
+![Do you use Rust at work?](/images/2022-02-RustSurvey/rust-at-work.png)
+
+As for why respondents are using Rust work, the top answer was that it allowed users "to build relatively correct and bug free software" with 96% of respondents agreeing with that statement. After correctness, performance (92%) was the next most popular choice. 89% of respondents agreed that they picked Rust at work because of Rust's much-discussed security properties. Perhaps surprisingly, 89% of respondents also chose Rust because their teams found it fun and enjoyable to program (though respondents did agree more strongly with their choice to pick Rust due to its security properties).
```
Comment (flip1995) [inline posts/2022-02-08-Rust-Survey-2021.md:37]:
```suggestion
As for why respondents are using Rust at work, the top answer was that it allowed users "to build relatively correct and bug free software" with 96% of respondents agreeing with that statement. After correctness, performance (92%) was the next most popular choice. 89% of respondents agreed that they picked Rust at work because of Rust's much-discussed security properties. Perhaps surprisingly, 89% of respondents also chose Rust because their teams found it fun and enjoyable to program (though respondents did agree more strongly with their choice to pick Rust due to its security properties).
```

File: posts/2022-02-08-Rust-Survey-2021.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+When asked what their biggest worries for the future of Rust are, the top answer was a fear that there will not be enough usage in industry (38%). Given that Rust continues to show strides in adoption at places of work, the community seems to be on a good path to overcoming this concern. 
+
+The next largest concern was that the language would become too complex (33%). This was combined with a relative small number of folks calling for additional features (especially for ones not already in the pipeline). 
+
+Finally, the third largest concern was that those working on Rust would not find the proper support they need to continue to develop the language and community in a healthy way. With the establishment of the Rust Foundation, support structures are coming into place that hopefully will address this point, but no doubt plenty of work is still ahead of us.
+
+## An Exciting Future
+
+2021 was arguably one of the most significant year's in Rust's history - with the establishment of the Rust foundation, the 2021 edition, and a larger community than ever, Rust seems to be a solid path as we head into the future.
```
Comment (flip1995) [inline posts/2022-02-08-Rust-Survey-2021.md:63]:
```suggestion
2021 was arguably one of the most significant year's in Rust's history - with the establishment of the Rust foundation, the 2021 edition, and a larger community than ever, Rust seems to be on a solid path as we head into the future. 
```

**Review by flip1995 — COMMENTED** [review]
Some typos I found.

I wonder about the survivor-ship bias of this survey. Do we have numbers for people who used Rust before but stopped using it (and the answers they gave)?

File: posts/2022-02-08-Rust-Survey-2021.md (hunk 4 above)
Comment (scottmcm) [reply posts/2022-02-08-Rust-Survey-2021.md:63]: Also it should be "most significant years in", no apostrophe.

**Comment by rylev** [comment]
@flip1995 survivorship bias is likely going to impact our results somewhat. Unfortunately, we don't have good data on how former users would answer some of these questions (we need to revisit what we ask former users for future surveys). Also, those who are former users who answered the survey are unlikely to be representative of all former users. That being said 75% of former users said that they answered the survey because they anticipate using Rust again in the future which I guess can provide some evidence that if you stop using the language, it's unlikely because you hated it. 🤷‍♀️

I think the best way to address this is to just bring more people's attention to it - perhaps by saying something about taking the numbers with a bit of a grain of salt. On the other hand, this is a casual blog post and doesn't claim to be a rigorous statistical study, so maybe that would be overkill. I'm not sure.

**Comment by flip1995** [comment]
Thanks for the further insights! 

> That being said 75% of former users said that they answered the survey because they anticipate using Rust again in the future

That's good to hear! Probably doesn't reflect the stopped-using-Rust group of people completely though.

I think this is something to look into further this year. I try to think about possible ways to approach this until the next survey comes up :)

> perhaps by saying something about taking the numbers with a bit of a grain of salt. On the other hand, this is a casual blog post and doesn't claim to be a rigorous statistical study

Yeah, I wouldn't necessarily mention it in the blog post. This was just a thought that came up when looking at those to-good-to-be-true numbers :smile:

**Comment by steveklabnik** [comment]
One thing that isn't in this post this year, but has always been helpful to talk about, is usage of nightly vs latest stable vs other versions. I don't know if we have that data or analyzed it, but it's in my top two or three things I am curious about from the survey.

Otherwise, great post! this is a tremendous amount of work, but is so useful. Thank you 🙏

**Comment by nrc** [comment]
> One thing that isn't in this post this year, but has always been helpful to talk about, is usage of nightly vs latest stable vs other versions. I don't know if we have that data or analyzed it, but it's in my top two or three things I am curious about from the survey.

That was intentional. We'll have a complete summary for the teams at least (and which I would like to make public too) which will include that data and other details. For the blog post we thought that the stable vs nightly split (and perceptions of stability, etc) were not as interesting to most users as they were a few years ago.

File: posts/2022-02-08-Rust-Survey-2021.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+We’d like to thank everyone who participated in this year’s survey with a special shout-out to those who helped translate the survey from English into other languages.
+
+Without further ado, let’s dive into the analysis!
+
+## A Global Community
+
+The Rust community continues to grow, with this survey having the largest number of complete survey responses (9354 respondents), exceeding last year's total by roughly 1500 responses.
+
+90% of respondents said that they use Rust for any purpose including learning (an all-time high!) with 5% stating they have in the past and 4% stating they have yet to use Rust.
```
Comment (Mark-Simulacrum) [inline posts/2022-02-08-Rust-Survey-2021.md:21]: Can you say more about 'all time high' here -- it's not clear to me why the percentage of respondents *using* Rust being high is a thing to celebrate. Don't we expect a high metric here?

It looks like this is mirroring language from previous years, but it feels like an odd thing to call out here. The particular sentence structure may also suggest that /learning/ is what is at an all time high, which is presumably incorrect? "learning" also feels like it's a little dangling? Are they just new users to Rust, or does learning here mean "formal" education (e.g., a CS degree).

File: posts/2022-02-08-Rust-Survey-2021.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+The survey was offered in 10 languages with 78% filling out the survey in English followed by Chinese (6%), German (4%), and French (3%). Despite English being the language most respondents completed the survey in, respondents hailed from all around the world. The United States was the country with the largest representation at 24% followed by Germany (12%), China (7%), and the U.K. (6%). In total 113 different countries and territories were represented through this survey!
+
+English, however, is not the language of choice for all Rustaceans with nearly 7% preferring not to use English as a language for technical communication. An additional 23% of respondents prefer another language in addition to English. The most common prefered languages (besides English) follow where Rustaceans live: Chinese, German, and French being the top 3. Japanese, Chinese, and Russian speakers were the most likely to prefer not to use English for technical conversation.
```
Comment (Mark-Simulacrum) [inline posts/2022-02-08-Rust-Survey-2021.md:25]: "follow where Rustaceans live: Chinese, German, and French being the top 3" 

This seems oddly ordered -- the previous paragraph suggests Germany has more respondents than China? Maybe we meant to say "the choice of language for filling out the survey"?

**Review by Mark-Simulacrum — COMMENTED** [review]
Looks great! I left a couple comments that seem good to resolve, but then let's set the date for either today or Monday to merge this.

File: posts/2022-02-08-Rust-Survey-2021.md (hunk 5 above)
Comment (rylev) [reply posts/2022-02-08-Rust-Survey-2021.md:21]: I think this is fair. We called this out as a thing to celebrate in previous posts, but I agree it's not really worth calling out.

As for "learning" - this refers to learning Rust in any capacity. I think we can just remove that bit.

File: posts/2022-02-08-Rust-Survey-2021.md (hunk 6 above)
Comment (rylev) [reply posts/2022-02-08-Rust-Survey-2021.md:25]: I added the term "roughly" since it's true that it doesn't follow the order exactly. Is that enough?

**Comment by rylev** [comment]
@Mark-Simulacrum @nrc I've addressed the feedback and adjusted the date to tomorrow (2022-02-15). Any remaining blockers?

File: posts/2022-02-08-Rust-Survey-2021.md (hunk 6 above)
Comment (Mark-Simulacrum) [reply posts/2022-02-08-Rust-Survey-2021.md]: Yeah, seems good.

**Review by Mark-Simulacrum — APPROVED** [review]

File: posts/2022-02-15-Rust-Survey-2021.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+75% of all Rust users say they can write production ready code, though 27% say that it is at times a struggle.
+
+![Rust expertise](/images/2022-02-RustSurvey/expertise.png)
+
+Overall, Rustaceans seem to be having a great time writing Rust with only 1% saying it isn't fun to use. Only a quarter of a percent find Rust doesn't have any real benefit over other programming languages.
+
+## Rust at Work
+
+Rust can now safely be classified as a language used by people in professional settings. 59% of respondents who currently use Rust, use it at least occasionally at work with 23% using Rust for the majority of their coding. This is a large increase over last year where 58% of respondents didn't use Rust at work compared with this year's 41%.
```
Comment (nikomatsakis) [inline posts/2022-02-15-Rust-Survey-2021.md:39]: I had to read this paragraph several times. It particularly confused me that it switched from "percent that do use rust" to "percent that don't" when doing a comparison. Suggested rewrite.

```suggestion
Rust can now safely be classified as a language used by people in professional settings. Of those respondents using Rust, 59% use it at least occasionally at work with 23% using Rust for the majority of their coding. This is a large increase over last year where only 42% of respondents used Rust at work.
```

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
Thanks for this! Nice easy read, clear narrative. Left one nit.

File: posts/2022-02-15-Rust-Survey-2021.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+Adopting Rust at work seems to follow a challenging but ultimately worthwhile path for a lot of Rustaceans. First, 83% of those who have adopted Rust at work found it to be "challenging". How much this is related to Rust itself versus general challenges with adopting a new programming language, however, is unclear. During adoption only 13% of respondents found the language was slowing their team down; which contrasts with the 82% who found that Rust helped their teams achieve their goals.
+
+After adoption, the costs seem to be worth it: only 1% of respondents did not find the challenge worth it while 79% said it definitely was. When asked if their teams were likely to use Rust again in the future, 90% agreed.
```
Comment (wesleywiser) [inline posts/2022-02-15-Rust-Survey-2021.md:47]: I wonder if we should re-arrange some of this to more explicitly state the progression path from challenging to helping achieve their goals to fun and enjoyable to program in. Something like:

```suggestion
Adopting Rust at work seems to follow a long but ultimately worthwhile path for a lot of Rustaceans. First, 83% of those who have adopted Rust at work found it to be "challenging". How much this is related to Rust itself versus general challenges with adopting a new programming language, however, is unclear. During adoption only 13% of respondents found the language was slowing their team down and 82% found that Rust helped their teams achieve their goals. 

After adoption, the costs seem to be justified: only 1% of respondents did not find the challenge worth it while 79% said it definitely was. When asked if their teams were likely to use Rust again in the future, 90% agreed. Finally, of respondents using Rust at work, 89% of respondents said their teams found it fun and enjoyable to program.

As for why respondents are using Rust at work, the top answer was that it allowed users "to build relatively correct and bug free software" with 96% of respondents agreeing with that statement. After correctness, performance (92%) was the next most popular choice. 89% of respondents agreed that they picked Rust at work because of Rust's much-discussed security properties. 
```

**Comment by joshtriplett** [comment]
The "rendered" link currently 404s.

**Comment by rylev** [comment]
> The "rendered" link currently 404s.

Fixed!

File: posts/2022-02-15-Rust-Survey-2021.md (hunk 8 above)
Comment (rylev) [reply posts/2022-02-15-Rust-Survey-2021.md:47]: Love it!

File: posts/2022-02-15-Rust-Survey-2021.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+The IDE experience (led through continued adoption and improvement of various tools like rust-analyzer, IntelliJ Rust, etc.) gets the prize for showing the most improvement:  56% found it has improved over the last year.
+
+However, compiler error messages received the most praise, with 90% approval of their current state. 🎉
+
+When asked what their biggest worries for the future of Rust are, the top answer was a fear that there will not be enough usage in industry (38%). Given that Rust continues to show strides in adoption at places of work, the community seems to be on a good path to overcoming this concern.
+
+The next largest concern was that the language would become too complex (33%). This was combined with a relative small number of folks calling for additional features (especially for ones not already in the pipeline).
+
+Finally, the third largest concern was that those working on Rust would not find the proper support they need to continue to develop the language and community in a healthy way. With the establishment of the Rust Foundation, support structures are coming into place that hopefully will address this point, but no doubt plenty of work is still ahead of us.
```
Comment (joshtriplett) [inline posts/2022-02-15-Rust-Survey-2021.md:65]: The previous two paragraphs had a percentage; could this one add a percentage as well?

**Comment by steveklabnik** [comment]
@nrc very disappointing :(
